### PR TITLE
storaged: New layout

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

**Comment by mvollmer** [comment]
@andreasn, feel free to play with this.

**Comment by mvollmer** [comment]
Tests will fail, I guess.

**Comment by mvollmer** [comment]
Depends on #4822, since that allows us to use simple "+/-" buttons for the sidebar.

**Comment by mvollmer** [comment]
The layout for plain disks (not raid or lvm) is wrong.

**Comment by mvollmer** [comment]
> The layout for plain disks (not raid or lvm) is wrong.

Fixed.

**Comment by stefwalter** [comment]
Storage failures on centos-7 and debian-unstable seem to be related.

**Comment by mvollmer** [comment]
> Storage failures on centos-7 and debian-unstable seem to be related.

I think this is unlikely.  The failure on debian looks like another instance of #4181.  I try to reproduce the centos-7 failure.

**Comment by mvollmer** [comment]
> I try to reproduce the centos-7 failure.

No luck.  The failure was during a resize, which isn't touched by this change.

**Comment by andreasn** [comment]
Looks good to me, ui-wise.

**Comment by dperpeet** [comment]
retriggering all tests because of elapsed time and other storage related code that was merged in the meantime

File: pkg/storaged/details.js [hunk 1]
```diff
@@ -1553,23 +1558,27 @@
...
-                                     Members: client.mdraids_members[mdraid.path].map(make_member),
-                                     DynamicMembers: (mdraid.Level != "raid0"),
-                                     Content: (block &&
-                                               mustache.render(content_tmpl,
-                                                               { Title: _("Content"),
-                                                                 path: block.path,
-                                                                 Entries: block_content_entries(block)
-                                                               }))
-                                   });
+            return [ mustache.render(mdraid_detail_tmpl,
```
Comment (dperpeet) [inline pkg/storaged/details.js]: Could you name these parts? Just having an array is difficult to read if you're not familiar with the code.
Right now these aren't documented and can only be understood by reading the use: https://github.com/cockpit-project/cockpit/pull/4854/files#diff-59bbb3cde23e86514ebe995dc75261d3R1747

Comment (mvollmer) [inline pkg/storaged/details.js]: > Could you name these parts? 

Good point, done.

**Comment by stefwalter** [comment]
@mvollmer @andreasn Do you have screenshots or a video (that we could reuse for the release notes)?

**Comment by mvollmer** [comment]
I have recorded a quick screencast: https://www.youtube.com/watch?v=vRdz75VGQPw

File: pkg/storaged/details.js (hunk 1 above)
Comment (dperpeet) [inline pkg/storaged/details.js]: awesome, so much more readable :)

**Comment by dperpeet** [comment]
failures are unrelated

**Pull request closed without merging**